src/pages/LandingPage.jsx

```diff
@@ -0,0 +1,84 @@
+// ============================================================================================
+// IMPORTS
+// ============================================================================================
+
+import { useContext, useState } from "react";
+import ResponsiveGrid from "../components/ResponsiveGrid";
+import { CoursesContext } from "../contexts/CoursesContext";
+import CourseCard from "../components/CourseCard";
+import Course from "../components/Course";
+import { useNavigate } from "react-router-dom";
+
+const serverURL = import.meta.env.VITE_SERVER_URL_ROOT;
+
+console.assert(
+  serverURL?.length > 0,
+  'Server URL not specified -- add "VITE_SERVER_URL_ROOT=<url>" to .env'
+);
+
+function LandingPage() {
+  // CONTEXT AND STATE
+  // context will have all the courses that will be displayed in the landing page
+  const { coursesInfo, setCoursesInfo } = useContext(CoursesContext);
+  // Course has the information for the course that would like to be displayed
+  const [course, setCourse] = useState(null);
+  // controlled input state that will help to filter the courses that are displayed
+  const [searchText, setSearchText] = useState("");
+  // state for the courses that will alter the courses displayed
+  const courses = coursesInfo?.Courses_data ? coursesInfo.Courses_data : [];
+  const [coursesDisplay, setCoursesDisplay] = useState(courses);
```

Comment:
coursesDisplay should be initialized based on courses directly. You might want to use useEffect to update coursesDisplay whenever coursesInfo changes.

Reply:
that's a great suggestion I will add the useEffect with courses as a dependency